File: Python/806. Number of Lines To Write String.py
class Solution:
    def numberOfLines(self, widths, S):
        """
        :type widths: List[int]
        :type S: str
        :rtype: List[int]
        """
        row, col = 1, 0
        for c in S:
            width = widths[ord(c) - ord('a')]
            if col + width > 100:
                row += 1
                col = width
            else:
                col += width
        return [row, col]

File: C++/806. Number of Lines To Write String.cpp
class Solution {
public:
    vector<int> numberOfLines(vector<int>& widths, string S) {
        int row=1,col = 0;
        for(auto c: S)
        {
            int num = widths[c-'a'];
            if(col+num > 100)
            {
                row += 1;
                col = num;
            }
            else
                col += num;
        }
        return {row, col};
    }
};